server/src/main/java/org/opensearch/action/search/SearchRequest.java

```diff
@@ -615,6 +624,43 @@ public void setPreFilterShardSize(int preFilterShardSize) {
         this.preFilterShardSize = preFilterShardSize;
     }
 
+    enum ParamValue {
+        TRUE,
+        FALSE,
+        UNSET
+    }
+
+    public static Boolean parseParamValue(Object str) {
+        if ("TRUE".equals(str)) {
+            return true;
+        } else if ("FALSE".equals(str)) {
+            return false;
+        } else {
+            return null;
+        }
+    }
```

Comment:
What is this giving us beyond a regular `Boolean`?

Also, why do we need the unset state anyway? Shouldn't we be `false` by default and only `true` if explicitly set? I don't see why we wouldn't just use a `boolean`.

Reply:
Ahh... the unset state means that we look at the cluster setting. See my other comment -- I disagree with the cluster setting. IMO, we shouldn't return phase_took unless it's explicitly requested.

Reply:
Yes, we need to distinguish undefined from true/false. `ParamValue` enum is unnecessary though. I have replaced it with Boolean.